developer-tools.md

```diff
@@ -246,23 +246,6 @@ Kubernetes, and more importantly, minikube have rapid release cycles, and point
 
 Apache Spark leverages GitHub Actions that enables continuous integration and a wide range of automation. Apache Spark repository provides several GitHub Actions workflows for developers to run before creating a pull request.
 
-<a name="github-workflow-tests"></a>
```

Comment:
Actually, I wonder if we can leave this here and point to it in the contributing guide?

Reply:
This process become obsolete after https://github.com/apache/spark/pull/32092 and is incorrect as it stands and hence this PR merges the new process to the Pull request section so it's clear for the new contributors.